compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -554,13 +582,23 @@ checkGroundType ty = do
 checkExpr' :: MonadGamma m => Expr -> Type -> m Type
 checkExpr' expr typ = do
   exprType <- typeOf expr
-  unless (alphaEquiv exprType typ) $
+  exprTypeX <- expandTypeSynonyms exprType
+  typX <- expandTypeSynonyms typ
+  unless (alphaEquiv exprTypeX typX) $
     throwWithContext ETypeMismatch{foundType = exprType, expectedType = typ, expr = Just expr}
   pure exprType
 
 checkExpr :: MonadGamma m => Expr -> Type -> m ()
 checkExpr expr typ = void (checkExpr' expr typ)
 
+-- | Check that a type synonym definition is well-formed.
```

Comment:
This is still missing the check for cyclic type synonyms. Let’s leave that for a separate PR but please add a TODO.

Reply:
we have this TODO already... In `checkModule` :
```-- TODO #3616: check type synonyms for cycles```

Reply:
:+1: